File: backend/Models/ShopBooking.js
let mysqlConfig = require("../Utilities/mysqlConfig");

let initialize = () => {
	mysqlConfig.getDB().query("create table IF NOT EXISTS shopBookings (shopId INT, date DATE, slotNumber INT, capacityLeft INT, tokensVerified INT DEFAULT 0, maxCapacity INT, PRIMARY KEY (shopId, date, slotNumber))");
	console.log("shopBookings created if not EXISTS")
}

module.exports = {
	initialize: initialize
}

File: token_system/lib/Entities/shop_booking.dart
class ShopBooking{
  int shopId;
  String date;
  int slotNumber;
  int capacityLeft;
  int tokensVerified;
  int maxCapacity;

  ShopBooking({
    this.shopId,
    this.date,
    this.slotNumber,
    this.capacityLeft,
    this.tokensVerified,
    this.maxCapacity
  });

  factory ShopBooking.fromJson(Map<String, dynamic> json, int shopId) {
    return ShopBooking(
      shopId: shopId,
      date: json['date'] as String,
      slotNumber: json['slotNumber'] as int,
      capacityLeft: json['capacityLeft'] as int,
      tokensVerified: json['tokensVerified'] as int,
      maxCapacity: json['maxCapacity'] as int
    );
  }
}